dace/transformation/estimator/enumeration/scoring_enumerator.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+""" This file implements the Enuerator class """
+
+from dace.transformation.subgraph import SubgraphFusion, helpers
+from dace.properties import make_properties, Property
+from dace.sdfg import SDFG, SDFGState
+from dace.sdfg.graph import SubgraphView
+
+from dace.transformation.estimator.enumeration import Enumerator
+import dace.sdfg.nodes as nodes
+
+from collections import deque, defaultdict, ChainMap
+from typing import Set, Union, List, Callable
+import itertools
+import warnings
+
+
+@make_properties
+class MapScoringEnumerator(Enumerator):
```

Comment:
is this class necessary?

Reply:
The idea was to make a parent class for all enumerators that rely on a score (-> performance model), e.g. connected and brute force.